src/main/java/seedu/us/among/commons/util/HeaderUtil.java

```diff
@@ -20,9 +20,12 @@
         HashMap<String, String> headerMap = new HashMap<>();
         for (Header header : headerSet) {
             String headerString = header.toString();
+            //to-do update trimming once handling of headers is updated on add
+            //trim leading and trailing brackets and quotations
+            headerString = headerString.substring(2, headerString.length() - 2);
```

Comment:
Just to confirm, Header's regex checks that a header is in the JSON format right? 

If not, such deletion before checking may lead to bugs.

Reply:
There's no check for header to be in json format. As an example, the header will be added in this format: `-h "Content-Type: application/json"`. That said, when retrieving the individual key and value, I noticed that this was what I got: `["Content-Type: application/json"]`. Hence, I trimmed the brackets and quotations to obtain the key and value. Ideally, the headers should be processed before being stored but until then this will still work :D